packages/headless/src/controllers/knowledge/generatedAnswer/headless-knowledge-generated-answer.test.ts

```diff
@@ -0,0 +1,285 @@
+import {
+  closeGeneratedAnswerFeedbackModal,
+  dislikeGeneratedAnswer,
+  likeGeneratedAnswer,
+  openGeneratedAnswerFeedbackModal,
+  registerFieldsToIncludeInCitations,
+  sendGeneratedAnswerFeedback,
+  setIsVisible,
+  updateResponseFormat,
+} from '../../../features/generated-answer/generated-answer-actions';
+import {
+  generatedAnswerAnalyticsClient,
+  GeneratedAnswerFeedback,
+} from '../../../features/generated-answer/generated-answer-analytics-actions';
+import {
+  GeneratedAnswerState,
+  getGeneratedAnswerInitialState,
+} from '../../../features/generated-answer/generated-answer-state';
+import {
+  buildMockSearchEngine,
+  MockedSearchEngine,
+} from '../../../test/mock-engine-v2';
+import {createMockState} from '../../../test/mock-state';
+import {
+  GeneratedAnswerProps,
+  GeneratedResponseFormat,
+} from '../../generated-answer/headless-generated-answer';
+import {buildKnowledgeGeneratedAnswer} from './headless-knowledge-generated-answer';
+
+jest.mock('../../../features/generated-answer/generated-answer-actions');
+jest.mock(
+  '../../../features/generated-answer/generated-answer-analytics-actions'
+);
+jest.mock('../../../features/search/search-actions');
+
+describe('knowledge-generated-answer', () => {
+  it('should be tested', () => {
+    expect(true).toBe(true);
+  });
+
+  let engine: MockedSearchEngine;
+
+  const createGeneratedAnswer = (props: GeneratedAnswerProps = {}) =>
+    buildKnowledgeGeneratedAnswer(
+      engine,
+      generatedAnswerAnalyticsClient,
+      props
+    );
+
+  const buildEngineWithGeneratedAnswer = (
+    initialState: Partial<GeneratedAnswerState> = {}
+  ) => {
+    const state = createMockState({
+      generatedAnswer: {
+        ...getGeneratedAnswerInitialState(),
+        ...initialState,
+      },
+    });
+    return buildMockSearchEngine(state);
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    engine = buildEngineWithGeneratedAnswer();
+  });
+
+  it('initializes', () => {
+    const generatedAnswer = createGeneratedAnswer();
+    expect(generatedAnswer).toBeTruthy();
+  });
+
```

Comment:
I would add a test to ensure it loads the answerApi reducer, and another test to assert that it dispatches the `updateConfigurationId` action with the correct answerConfigurationId as a payload upon initialization.